### PR TITLE
Reformat service limits to have max_route_location per costing

### PR DESCRIPTION
**Pull request by kdiluca**
- also added matrix limits (not yet being used in master)

### CODE REVIEW

**Comment by dnesbitt61** [comment]
:shipit:

File: valhalla.json [hunk 1]
```diff
@@ -99,18 +99,59 @@
...
+      "max_route_locations": 50
+    },
+    "multimodal": {
+      "max_distance": 500000.0,
+      "max_route_locations": 50
+    },
     "max_shape": 750000,
     "min_resample": 10.0 
   },
+  "matrix_limits": {
```
Comment (kevinkreiser) [inline valhalla.json]: why is there a separate limits for all of the costings under matrix? matrix uses simple dijsktra for all modes of transport, and doesnt really need different numbers of locations or distance per mode does it? i thought we were going to have each of the three matrix types under the main limits node and just have each have a distance and location limit?

Comment (dnesbitt61) [inline valhalla.json]: Good point. I think a single matrix_limits sections with max_distance and max_locations for each type makes sense. max_distance will mean something a little different, so maybe we should use a different name, but not sure what that might be.

Comment (kevinkreiser) [inline valhalla.json]: thats exactly what i was thinking with respect to the name but also couldnt think of a decent suggestion!

**Pull request merged**